src/instrument.ts

```diff
@@ -136,12 +139,20 @@ export function startTrace<T>(
     config = {} as traceConfig;
   }
 
-  ensureInitialContextManagerSet();
+  let parentSpanId: string | undefined;
+  if (debugTraces) {
```

Comment:
can we move `debugTraces` changes to a separate PR ? This should not be related to the issues we found out the other day ?

Reply:
It was super necessary to get out this PR but sure I'll remove the debugTraces code from this PR.